hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/commit/UpsertPartitioner.java

```diff
@@ -146,7 +146,7 @@ private int addUpdateBucket(String partitionPath, String fileIdHint) {
    * @return smallFiles not in clustering
    */
   private List<SmallFile> filterSmallFilesInClustering(final Set<String> pendingClusteringFileGroupsId, final List<SmallFile> smallFiles) {
-    if (this.config.isClusteringEnabled()) {
+    if (this.config.isClusteringEnabled() || !pendingClusteringFileGroupsId.isEmpty()) {
```

Comment:
If the clustering is not enabled, can the `pendingClusteringFileGroupsId ` set be non-empty? Would be great if we can add a test to replicate the scenario.

Reply:
Hi @codope Thanks for your view. 

As I mentioned above.  If the clustering is not enabled during insert but there is an offline clustering plan job is running for this table, the `pendingClusteringFileGroupsId` set could be non-empty too. 

I just add a UT named `testUpsertPartitionerWithSmallFileHandlingAndClusteringPlan` to prove it. I set `withClusteringConfig(HoodieClusteringConfig.newBuilder().withInlineClustering(false).withAsyncClustering(false).build())` during ingestion and create a clustering plan manually out of this ingestion scope. 

As you you can see, If the condition is `this.config.isClusteringEnabled()` only, there will be two small slices can be inserted including file slice 1 in pending clustering plan. 

If the condition is `!pendingClusteringFileGroupsId.isEmpty()`, there will be only one small slice can be used and file slice 1 is filtered.

Reply:
Got it. Thanks @zhangyue19921010